packages/jspsych/src/ProgressBar.ts

```diff
@@ -14,7 +17,13 @@ export class ProgressBar {
   /** Adds the progress bar HTML code into `this.containerElement` */
   private setupElements() {
     this.messageSpan = document.createElement("span");
-    this.messageSpan.innerHTML = this.message;
+
+    if (Object.prototype.toString.call(this.message) === "[object Function]") {
+      this.messageSpan.innerHTML = (this.message as (progress: number) => string)(0);
+    } else {
+      // Progress starts at 0 when experiment commences
+      this.messageSpan.innerHTML = this.message as string;
+    }
```

Comment:
I'd prefer extracting this bit into a private `updateMessage()` method so we don't need to duplicate parts of it in the `update()` method.

Reply:
Fixed this and also the function type check in 4dbdbc0.